simplifier/operations/boolean.py

```diff
@@ -196,17 +199,18 @@ def accept(self, visitor: WorldObjectVisitor[T]) -> T:
     def fold(self, keep_form: bool = False):
         """Fold the constants."""
         constants = self.constants
-        if len(constants) > 1 and len(set((self._interpret(constant) for constant in constants))) == 1:
-            for constant in constants[1:]:
-                self.remove_operand(constant)
+        if 2 <= len(constants) < len(self.operands):
+            if len(set((self._interpret(constant) for constant in constants))) == 1:
+                for constant in constants[1:]:
+                    self.remove_operand(constant)
 
     def _get_splitted_operands(self) -> Iterator[Relation]:
         operands = self.operands
         for op1, op2 in zip(operands[:-1], operands[1:]):
             yield self.world.bool_equal(op1, op2)
 
 
-class Unequal(NonEqualRelation, UnsignedRelation):
+class Unequal(NonEqualRelation, UnsignedRelation, CommutativeOperation):
```

Comment:
Same as for `Equal`, I am not sure whether we want the `CommutativeOperation` here.

Reply:
```suggestion
class Unequal(NonEqualRelation, UnsignedRelation):
```
I will remove the CommutativeOperation for now.